Lemon Jsonparser step1

puuurm opened the pull request:
InputView: 사용자로 부터 JSON 형태의 데이터를 입력받아 String을 출력한다.
JSONAnalyzer: JSON 형태의 스트링을 분석하여
1) 배열 형식이고
2) Bool, String, Int 타입이 충족되면
JSONObject를 리턴한다.
OutputView: JSONObject 정보를 출력한다.

puuurm committed:
Add InputView structure

Add InputView structure to input JSON data string, and input string."

puuurm committed:
Create a JSONObject structure to manage the json object

puuurm committed:
Create JSONAnalyzer to convert json data of String type to jsonObject instance

puuurm committed:
Create an OutputView structure that outputs the result of analyzing the json object

puuurm committed:
Edit the main function to output the analysis result by inputting json string and creating json object

puuurm commented:
커밋하고 보니까 커밋 메세지가 너무 기네요.. 앞으로는 커밋 제목은 간결하게 줄이겠습니다.

godrm commented on JSONParser/JSONParser/InputView.swift:
```diff
@@ -0,0 +1,23 @@
+
+struct InputView {
+    
+    func readInput() throws -> String? {
```
InputView 객체의 readInput() 메소드 이름은 Input처리하는 객체의 메소드라서 Input을 안 붙여도 될 것 같습니다.
InputError도 InputView 에서 발생하는 에러라면 InputView 내부에 nested 시키면 좋겠습니다. 
하나의 네임스페이스(namespace)로 접근할 때는 중복되는 단어를 제거해주세요.

godrm commented on JSONParser/JSONParser/main.swift:
```diff
@@ -8,3 +8,23 @@
+        // 단순한 JSON 배열(array) 문자열을 입력하면 내부 요소들을 분석해서 배열로 저장하도록 구현한다.
+        
+        // 입력 받기
+        guard let jsonString = try inputView.readInput() else { break }
```
다른 객체들은 인스턴스를 만들지 않았는데 inputView만 인스턴스를 만든 이유가 있나요?

puuurm committed:
Fix JSONAnalyzer error handling

Move error enumeration from InputView structure to JSONAnalyzer structure.
Change the name of the read method of the InputView structure and change it to a static method.

puuurm commented:
Error handling을 InputView 구조체에서 JSONAnalyzer로 이동
InputView의 read 메소드를 static 메소드로 변경

puuurm commented:
입력할 때 생기는 에러가 아닌, Parsing 하는 과정에서 생기는 에러이므로 JSONAnalyzer로 이동하였습니다.

godrm commented:
커밋 메시지는 첫 줄은 요약을 쓰고, 상세 내용은 아래줄에 쓰는 방식도 있습니다.
다음 단계부터 구조적으로 어떤 객체 관계를 갖도록 할지 설명을 해주세요. 
그림을 그려도 좋고, 그냥 말로 설명해도 됩니다. 머지하겠습니다.
